stuffer/s2n_stuffer.c

```diff
@@ -37,9 +37,15 @@ bool s2n_stuffer_is_valid(const struct s2n_stuffer* stuffer)
 
 bool s2n_stuffer_reservation_is_valid(const struct s2n_stuffer_reservation* reservation)
 {
-    return S2N_OBJECT_PTR_IS_READABLE(reservation) &&
-           s2n_stuffer_is_valid(reservation->stuffer) &&
-           S2N_MEM_IS_WRITABLE(reservation->stuffer->blob.data + reservation->write_cursor, reservation->length);
+    if (!S2N_OBJECT_PTR_IS_READABLE(reservation)) {
```

Comment:
I'd add a comment why we're doing this, link to the CBMC issue, and add a tracking issue to roll back when CBMC can handle common subexpression elimination.

Reply:
Implemented as suggested. I also tagged the CBMC issue to track this performance improvement in the s2n issue https://github.com/awslabs/s2n/issues/2290.